yt/frontends/swift/io.py

```diff
@@ -0,0 +1,168 @@
+"""
+Swift data-file handling function
+
+
+
+
+"""
+from __future__ import print_function
+
+#-----------------------------------------------------------------------------
+# Copyright (c) 2018, yt Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+from yt.utilities.on_demand_imports import _h5py as h5py
+import numpy as np
+
+from yt.frontends.sph.io import \
+    IOHandlerSPH
+
+class IOHandlerSwift(IOHandlerSPH):
+    _dataset_type = "swift"
+
+    def __init__(self, ds, *args, **kwargs):
+        super(IOHandlerSwift, self).__init__(ds, *args, **kwargs)
+
+    def _read_fluid_selection(self, chunks, selector, fields, size):
+        raise NotImplementedError
+
+    def _read_particle_coords(self, chunks, ptf):
+        # This will read chunks and yield the results.
+        chunks = list(chunks)
+        data_files = set([])
+        for chunk in chunks:
+            for obj in chunk.objs:
+                data_files.update(obj.data_files)
+        for data_file in sorted(data_files, key=lambda x: x.filename):
+            si, ei = data_file.start, data_file.end
+            f = h5py.File(data_file.filename, "r")
+            # This double-reads
+            for ptype, field_list in sorted(ptf.items()):
+                if data_file.total_particles[ptype] == 0:
+                    continue
+                x = f["/%s/Coordinates" % ptype][si:ei, 0].astype("float64")
+                y = f["/%s/Coordinates" % ptype][si:ei, 1].astype("float64")
+                z = f["/%s/Coordinates" % ptype][si:ei, 2].astype("float64")
```

Comment:
Can't you do this by broadcasting and only do one I/O for the full coordinates array? I don't know offhand if that's actually any faster or more memory conservative, this may be fine.

Reply:
I would advocate for `x, y, z = f["/%s/Coordinates" % ptype][si:ei].T`. I am unsure if the `astype` conversion should be necessary as we store them in the HDF5 as doubles anyway.